File: app/src/main/java/com/asthmapp/android/features/learn/LearnNewsItem.kt
package com.asthmapp.android.features.learn

import com.asthmapp.android.R
import com.asthmapp.android.appContext
import com.asthmapp.features.learn.entity.LearnNews

sealed class LearnNewsItem {

    data class Video(private val video: LearnNews.Video) : LearnNewsItem() {

        val title = video.title
        val link = video.link
        val imageLink = video.imageLink
        val durationString = video.durationSeconds.toDurationString()

        private fun Int.toDurationString(): String {
            val seconds = (this % 60).toString().padStart(2, '0')
            val minutes = (this / 60).toString().padStart(2, '0')
            return "$minutes:$seconds"
        }
    }

    data class Article(private val article: LearnNews.Article) : LearnNewsItem() {

        val title = article.title
        val link = article.link
        val imageLink = article.imageLink
        val readDurationString = article.readTimeSeconds.toReadDurationString()

        private fun Int.toReadDurationString() = appContext.getString(R.string.article_read, this / 60)
    }
}

File: ios/Asthm App/Application/features/learn/entity/LearnNewsItem.swift
//
//  LearnNewsItem.swift
//  Asthm App
//
//  Created by Den Matiash on 04.01.2021.
//  Copyright © 2021 xorum.io. All rights reserved.
//

import Foundation
import common

enum LearnNewsItem {
    
    struct Video {
        
        let title: String
        let link: String
        let imageLink: String
        let durationString: String
        
        init(_ video: LearnNews.Video) {
            title = video.title
            link = video.link
            imageLink = video.imageLink
            durationString = video.durationSeconds.toDurationString()
        }
    }
    
    struct Article {
        
        let title: String
        let link: String
        let imageLink: String
        let readDurationString: String
        
        init(_ article: LearnNews.Article) {
            title = article.title
            link = article.link
            imageLink = article.imageLink
            readDurationString = article.readTimeSeconds.toReadDurationString()
        }
    }
    
    case video(Video)
    case article(Article)
    case emergency
    case search
}

fileprivate extension Int32 {

    func toDurationString() -> String {
        let seconds = self % 60
        let minutes = self / 60
        return String(format: "%02d:%02d", minutes, seconds)
    }
    
    func toReadDurationString() -> String {
        let minutes = self / 60
        return "article_read".localizedFormat(args: minutes)
    }
}
